app/controllers/operations_controller.rb

```diff
@@ -0,0 +1,47 @@
+class OperationsController < ApplicationController
+  before_action :redirect_unauthenticated_user_to_custom_page
+
+  def index
+    if params[:category_id].present?
+      @category = Category.find(params[:category_id])
+      @operations = current_user.operations.includes(:category).where(category: @category)
+      @total_amount = @operations&.sum(:amount) || 0
+    else
+      @operations = current_user.operations
+    end
+    @page_name = 'Transactions'
+  end
```

Comment:
- [ ] Great job working on the functionality and the design but I think it would be great if you fix the error that shows up when press on so I can be able to see the operations page 🤩✨💯

![image](https://user-images.githubusercontent.com/58553711/231631123-3ae2b1e9-a22b-44a9-ad34-2bbf1da0bf20.png)